evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -673,9 +673,14 @@ def test_automl_time_series_classification_threshold(mock_binary_fit, mock_binar
     automl.search()
     assert isinstance(automl.data_splitter, TimeSeriesSplit)
     if objective == 'Log Loss Binary':
-        mock_optimize_threshold.assert_not_called()
-        assert automl.best_pipeline.threshold is None
-        mock_split_data.assert_not_called()
+        if not optimize:
+            mock_optimize_threshold.assert_not_called()
+            assert automl.best_pipeline.threshold is None
+            mock_split_data.assert_not_called()
+        else:
```

Comment:
Check that we optimize the threshold even if the main objective isn't optimizable